File: src/auth/auth.controller.ts
import { AuthService } from './auth.service';
import { Body, Controller, HttpCode, HttpStatus, Post } from '@nestjs/common';
import { SignInDataDto, SignInRequestDto } from './dto/signin-user.dto';

@Controller('auth')
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @HttpCode(HttpStatus.OK)
  @Post('signin')
  async signin(@Body() dto: SignInRequestDto) {
    const user = await this.authService.validateUser(dto.email, dto.password);
    const jwtToken = this.authService.generateJwt(user);

    const data: SignInDataDto = {
      userId: Number(user.user_id),
      email: user.email,
      role: user.role,
    };

    return { status: 'SUCCESS', data: data, token: jwtToken };
  }
}

File: src/main/java/com/ai/dailyReport/auth/controller/AuthController.java
package com.ai.dailyReport.auth.controller;

import com.ai.dailyReport.auth.dto.SignInRequestDto;
import com.ai.dailyReport.auth.dto.SignInResponseDto;
import com.ai.dailyReport.auth.jwt.JwtTokenProvider;
import com.ai.dailyReport.domain.entity.User;
import com.ai.dailyReport.domain.repository.UserRepository;
import com.ai.dailyReport.auth.service.AuthService;
import com.ai.dailyReport.common.exception.UnauthorizedException;

import lombok.RequiredArgsConstructor;


import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

import jakarta.validation.Valid;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {
    
    private final AuthService authService;
    private final JwtTokenProvider jwtTokenProvider;
    private final UserRepository userRepository;
    
    @PostMapping("/signin")
    public ResponseEntity<SignInResponseDto> signin(@Valid @RequestBody SignInRequestDto requestDto) {
        User user = authService.validateUser(requestDto.getEmail(), requestDto.getPassword());
        String jwtToken = jwtTokenProvider.generateToken(user.getEmail(), user.getRole());
        
        SignInResponseDto response = SignInResponseDto.builder()
            .status("SUCCESS")
            .data(SignInResponseDto.UserData.builder()
                .userId(user.getUserId())
                .email(user.getEmail())
                .role(user.getRole())
                .name(user.getName())
                .build())
            .token(jwtToken)
            .build();
            
        return ResponseEntity.ok(response);
    }


}